prusti-viper/src/encoder/backtranslation.rs

```diff
@@ -0,0 +1,457 @@
+                None => Entry::Unit,
+                Some((rust_name, vir_name, typ)) => {
+                    let result_vir_name = String::from("Result()");
+                    let opt_entry = silicon_entries.get(&result_vir_name);
+                    backtranslator.backtranslate_entry(
+                        typ, 
+                        opt_entry, 
+                        result_vir_name,
+                        silicon_entries,
+                    )
+                }
+            }
+        };
+        
+        Counterexample::Success {result, args, entries, is_pure: backtranslator.is_pure}
+    } else {
+        Counterexample::Failure(String::from("no counterexample generated"))
+    }
+}
+
+
+impl<'tcx> Backtranslator<'tcx>{
+    fn backtranslate_entry(
+        &self,
+        typ: Ty<'tcx>, 
+        sil_entry: Option<&ModelEntry>, 
+        vir_name: String,
+        silicon_ce_entries: &HashMap<String, ModelEntry>,
+    ) -> Entry {
+        match typ.kind(){
+            ty::TyKind::Bool => {
+                match sil_entry{
+                    Some(ModelEntry::LitBoolEntry(value)) => Entry::BoolEntry { value: *value },
+                    Some(ModelEntry::RefEntry(name, map)) => {
+                        let entry = map.get(&String::from("val_bool"));
+                        if let Some(ModelEntry::LitBoolEntry(value)) = entry {
+                            Entry::BoolEntry {value: *value}
+                        } else {
+                            Entry::UnknownEntry
+                        }   
+                    },
+                    _ => Entry::UnknownEntry
+                }
+            },
+            ty::TyKind::Int(_) | ty::TyKind::Uint(_) => {
+                let opt_value = self.backtranslate_int(sil_entry);
+                if let Some(value) = opt_value {
+                    Entry::IntEntry { value }
+                } else {
+                    Entry::UnknownEntry
+                }
+            },
+            ty::TyKind::Char => {
+                let opt_value = self.backtranslate_int(sil_entry);
+                if let Some(value) = opt_value {
+                    let val_t = u8::try_from(value);
+                    if let Ok(v) = val_t {
+                        return Entry::CharEntry{ value: v as char}
+                    } 
+                }
+                Entry::UnknownEntry
+            },
+            ty::TyKind::Ref(_, typ, _) => {
+                if let Some(ModelEntry::RefEntry(name, map)) = sil_entry {
+                    let entry = map.get(&String::from("val_ref"));
+                    let new_vir_name = format!("{}.val_ref", vir_name);
+                    let rec_entry = self.backtranslate_entry(
+                        typ, 
+                        entry, 
+                        new_vir_name,
+                        silicon_ce_entries, 
+                    );
+                    Entry::RefEntry {el: Box::new(rec_entry)} 
```

Comment:
```suggestion
                    Entry::RefEntry {el: box ec_entry}
```
(and in other places)